dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ResourcePolicyRestRepository.java

```diff
@@ -52,9 +95,249 @@ public ResourcePolicyRest findOne(Context context, Integer id) {
+     * 
+     * @param epersonUuid
+     *            mandatory, the uuid of the eperson that benefit of the policy
+     * @param resourceUuid
+     *            optional, limit the returned policies to the ones related to the specified resource
+     * @param pageable
+     *            contains the pagination information
+     *
+     * @return It returns the list of explicit matching resource policies, no inherited or broader resource policies
+     *         will be included in the list nor policies derived by groups' membership
+     */
+    @PreAuthorize("hasPermission(#epersonUuid, 'EPERSON', 'READ')")
+    @SearchRestMethod(name = "eperson")
+    public Page<ResourcePolicyRest> findByEPerson(@Parameter(value = "uuid", required = true) UUID epersonUuid,
+                                @Parameter(value = "resource", required = false) UUID resourceUuid, Pageable pageable) {
+
+        List<ResourcePolicy> resourcePolisies = null;
+        int total = 0;
+        try {
+            Context context = obtainContext();
+            EPerson eperson = epersonService.find(context, epersonUuid);
+            if (eperson == null) {
+                return null;
+            }
+            if (resourceUuid != null) {
+                resourcePolisies = resourcePolicyService.findByEPersonAndResourceUuid(context, eperson, resourceUuid,
+                        Math.toIntExact(pageable.getOffset()),
+                        Math.toIntExact(pageable.getOffset() + pageable.getPageSize()));
+                total = resourcePolicyService.countResourcePoliciesByEPersonAndResourceUuid(context,
+                        eperson, resourceUuid);
+            } else {
+                resourcePolisies = resourcePolicyService.findByEPerson(context, eperson,
+                        Math.toIntExact(pageable.getOffset()),
+                        Math.toIntExact(pageable.getOffset() + pageable.getPageSize()));
+                total = resourcePolicyService.countByEPerson(context, eperson);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+        return converter.toRestPage(resourcePolisies, pageable, total, utils.obtainProjection(true));
+    }
+
+    /**
+     * Find the resource policies matching uuid of the group and/or the ones specified resource object
+     * 
+     * @param groupUuid
+     *            mandatory, the uuid of the group that benefit of the policy
+     * @param resourceUuid
+     *            optional, limit the returned policies to the ones related to the specified resource
+     * @param pageable
+     *            contains the pagination information
+     * 
+     * @return It returns the list of explicit matching resource policies, no inherited or broader resource policies
+     *         will be included in the list nor policies derived by groups' membership
+     */
+    @PreAuthorize("hasPermission(#groupUuid, 'GROUP', 'READ')")
```

Comment:
I think this should likely be checking for GROUP **WRITE** (or maybe even ADMIN) permissions.  

I'm noticing that while *unauthenticated*, I can view **all the resource policies** of the Anonymous group via this REST call: `http://localhost:8080/server/api/authz/resourcepolicies/search/group?uuid=[uuid-of-Anonymous]` 

This seems like a security issue to me.

Reply:
We have had a similar discussion during the definition of the REST contract see https://github.com/DSpace/Rest7Contract/pull/87#discussion_r343754096
essentially we allow people to know what they are allowed to do.

The current contract says that the search method is available to member of the group and this actually mean READ permission. Maybe in future we could find an use case to give READ permission over a group without be a member of such group but this is not a feature that exists right now.
If you prefer to deal with this difference we should hardcode this check in the repository search method as it is not possible to express the "isMemberOf" concept as a DSpaceRestPermission (READ, WRITE, etc)

Reply:
this is the related test https://github.com/DSpace/DSpace/pull/2604/files#diff-3a5daad2fc92c374f068f92a31c36ec7R857

Reply:
@abollini : I understand your point.  However, I'm still a little concerned about making *every policy* related to the `Anonymous` group **publicly readable.**  I wonder if we should leave this `@PreAuthorize` as-is, but implement a special (hardcoded) check for the Anonymous group (to ensure it is only visible to authenticated users, perhaps)?  I don't like the idea of having any resource policies being publicly readable....at the minimum you need to have an authenticated account in the system.

Reply:
ok, implemented. A test has been added for this use case
https://github.com/DSpace/DSpace/pull/2604/commits/434774f1f650683f87366c26b1a373aa18a9c64c

it was failing with the original code https://travis-ci.org/DSpace/DSpace/builds/646783645?utm_source=github_status&utm_medium=notification
and fixed now  https://travis-ci.org/DSpace/DSpace/builds/646784796?utm_source=github_status&utm_medium=notification

Here the PR to sync the contract too https://github.com/DSpace/Rest7Contract/pull/101